prov/sm2/src/sm2.h

```diff
@@ -282,7 +276,7 @@ static inline uint64_t sm2_rx_cq_flags(uint32_t op, uint64_t rx_flags,
 				       uint16_t op_flags)
 {
 	rx_flags |= ofi_rx_cq_flags(op);
-	if (op_flags & SM2_REMOTE_CQ_DATA)
+	if (op_flags & FI_REMOTE_CQ_DATA)
 		rx_flags |= FI_REMOTE_CQ_DATA;
```

Comment:
this function was here to translate SM* flags into FI_* flags. now that you're saving the flags and just passing the originals around, let's change this into a single bitwise operation to only save the completion flags you care about - 
`rx_flags |= ofi_rx_cq_flags(op) | (op_flags & FI_REMOTE_CQ_DATA);`

Reply:
this change causes a hang... will debug tomorrow.

Reply:
Need rx_flags |=